WorkInProgress/Administration/xASL_adm_CoregisterAtlas.m

```diff
@@ -0,0 +1,131 @@
+matlabbatch{1}.spm.tools.dartel.warp1.settings.param(6).K = 6;
+matlabbatch{1}.spm.tools.dartel.warp1.settings.param(6).template = {fullfile(pathExploreASL, 'External', 'SPMmodified', 'toolbox', 'cat12', 'templates_volumes' ,'Template_6_IXI555_MNI152.nii')};
+matlabbatch{1}.spm.tools.dartel.warp1.settings.optim.lmreg = 0.01;
+matlabbatch{1}.spm.tools.dartel.warp1.settings.optim.cyc = 3;
+matlabbatch{1}.spm.tools.dartel.warp1.settings.optim.its = 3;
+
+spm_jobman('run',matlabbatch);
+%% Transform the template to the correct template, just as a test
+matlabbatch = [];
+matlabbatch{1}.spm.util.defs.comp{1}.dartel.flowfield = {fullfile(pathTPM, pathTPMAtlasDir, ['u_ws' pathTPMAtlasFile])};
+matlabbatch{1}.spm.util.defs.comp{1}.dartel.times = [1 0];
+matlabbatch{1}.spm.util.defs.comp{1}.dartel.K = 6;
+matlabbatch{1}.spm.util.defs.comp{1}.dartel.template = {fullfile(pathExploreASL, 'External', 'SPMmodified', 'toolbox', 'cat12', 'templates_volumes' ,'Template_6_IXI555_MNI152.nii')};
+matlabbatch{1}.spm.util.defs.out{1}.pull.fnames = {fullfile(pathTPM, pathTPMAtlasDir, ['ws' pathTPMAtlasFile])};
+matlabbatch{1}.spm.util.defs.out{1}.pull.savedir.saveusr = {pathTPM};
+matlabbatch{1}.spm.util.defs.out{1}.pull.interp = 4;
+matlabbatch{1}.spm.util.defs.out{1}.pull.mask = 1;
+matlabbatch{1}.spm.util.defs.out{1}.pull.fwhm = [0 0 0];
+matlabbatch{1}.spm.util.defs.out{1}.pull.prefix = 'o';
+
+spm_jobman('run',matlabbatch);
+
+%% Transform the atlases
+% This has to be done label per label
+
+% We load atlas and count all labels
+imAtlas = xASL_io_Nifti2Im(fullfile(pathTPM, 'AAL3', 'AAL3v1_1mm.nii'));
+max(imAtlas(:))
+imAtlas = xASL_io_Nifti2Im(fullfile(pathTPM, 'WMPM', 'WMPM_Type_III.nii'));
+max(imAtlas(:))
+
+% Nearest neighbor transformation
+
+matlabbatch = [];
+matlabbatch{1}.spm.util.defs.comp{1}.id.space = {fullfile(pathExploreASL, 'External', 'SPMmodified', 'toolbox', 'cat12', 'templates_volumes' ,'Template_6_IXI555_MNI152.nii')};
+matlabbatch{1}.spm.util.defs.comp{2}.dartel.flowfield = {fullfile(pathTPM, pathTPMAtlasDir, ['u_ws' pathTPMAtlasFile])};
+matlabbatch{1}.spm.util.defs.comp{2}.dartel.times = [1 0];
+matlabbatch{1}.spm.util.defs.comp{2}.dartel.K = 6;
+matlabbatch{1}.spm.util.defs.comp{2}.dartel.template = {fullfile(pathExploreASL, 'External', 'SPMmodified', 'toolbox', 'cat12', 'templates_volumes' ,'Template_6_IXI555_MNI152.nii')};
+matlabbatch{1}.spm.util.defs.out{1}.pull.fnames = {fullfile(pathTPM, 'AAL3', 'AAL3v1_1mm.nii')
+	                                               fullfile(pathTPM, pathTPMAtlasDir, pathTPMAtlasFile)
+												   fullfile(pathTPM, 'WMPM', 'WMPM_Type_III.nii')};
+matlabbatch{1}.spm.util.defs.out{1}.pull.savedir.saveusr = {pathTPM};
+matlabbatch{1}.spm.util.defs.out{1}.pull.interp = 0;
+matlabbatch{1}.spm.util.defs.out{1}.pull.mask = 1;
+matlabbatch{1}.spm.util.defs.out{1}.pull.fwhm = [0 0 0];
+matlabbatch{1}.spm.util.defs.out{1}.pull.prefix = 'o';
+
+spm_jobman('run',matlabbatch);
```

Comment:
If you say that this gives better alignment, I trust you. It makes sense resolution-wise, but the question is if the parcellation—pvGM varies between atlases, you create more bias the more non-linear you register based on the pvGM (and not based on the parcellation).

Also, if you want to do it non-linear, isn't Geodesic Shooting better?

But these are minor comments of course.

Usually, the difference between parcellations is minor and can be described only with affine. Did you test that this what not sufficient before going for what you thought of to be more optimal?